src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -1,14 +1,368 @@
+    auto context = openassetio::Context::make();
+
+    GIVEN("manager plugin successfully resolves a single entity reference") {
+      const openassetio::EntityReference ref = openassetio::EntityReference{"testReference"};
+      const openassetio::EntityReferences refs = {ref};
+
+      const openassetio::TraitsDataPtr expected = openassetio::TraitsData::make();
+      expected->addTrait("aTestTrait");
+
+      // With success callback side effect
+      REQUIRE_CALL(*mockManagerInterfacePtr, resolve(refs, traits, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_5(0, expected));
+
+      WHEN("resolve is called with default errorPolicyTag") {
+        const openassetio::TraitsDataPtr actual = manager->resolve(ref, traits, context);
+        THEN("returned TraitsData is as expected") { CHECK(expected.get() == actual.get()); }
+      }
+      WHEN("resolve is called with kException errorPolicyTag") {
+        const openassetio::TraitsDataPtr actual = manager->resolve(
+            ref, traits, context, openassetio::BatchElementErrorPolicyTag::kException);
+        THEN("returned TraitsData is as expected") { CHECK(expected.get() == actual.get()); }
+      }
+      WHEN("resolve is called with kVariant errorPolicyTag") {
+        std::variant<openassetio::TraitsDataPtr, openassetio::BatchElementError> actual =
+            manager->resolve(ref, traits, context,
+                             openassetio::BatchElementErrorPolicyTag::kVariant);
+        THEN("returned variant contains the expected TraitsData") {
+          CHECK(std::holds_alternative<openassetio::TraitsDataPtr>(actual));
+          auto actualVal = std::get<openassetio::TraitsDataPtr>(actual);
+          CHECK(expected.get() == actualVal.get());
+        }
+      }
+    }
+    GIVEN("manager plugin successfully resolves multiple entity references") {
+      const openassetio::EntityReferences refs = {openassetio::EntityReference{"testReference1"},
+                                                  openassetio::EntityReference{"testReference2"},
+                                                  openassetio::EntityReference{"testReference3"}};
+
+      const openassetio::TraitsDataPtr expected1 = openassetio::TraitsData::make();
+      expected1->addTrait("aTestTrait1");
+      const openassetio::TraitsDataPtr expected2 = openassetio::TraitsData::make();
+      expected2->addTrait("aTestTrait2");
+      const openassetio::TraitsDataPtr expected3 = openassetio::TraitsData::make();
+      expected3->addTrait("aTestTrait3");
+      std::vector<openassetio::TraitsDataPtr> expectedVec{expected1, expected2, expected3};
+
+      // With success callback side effect
+      REQUIRE_CALL(*mockManagerInterfacePtr, resolve(refs, traits, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_5(2, expectedVec[2]))
+          .LR_SIDE_EFFECT(_5(0, expectedVec[0]))
+          .LR_SIDE_EFFECT(_5(1, expectedVec[1]));
```

Comment:
[fix] Clever. Can we comment/make part of the GIVEN/WHEN/THEN as this is by design and checking critical behaviour of out-of-order CBs.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/864/commits/fff3ae8e934d5a6162944800955bbb7a6d571e60

What i've done here is added a fully seperate case specifically to test the out-of-order resolves.
It's duplication, but I think it gives the advantage of being to differentiate between a general test failure, (even in-order,) and a failure caused specifically by ordering expectations, which I expect might come up.